lib/plugins/aws/deploy/compile/events/schedule/index.js

```diff
@@ -54,19 +76,28 @@ class AwsCompileScheduledEvents {
 
             const normalizedFunctionName = functionName[0].toUpperCase() + functionName.substr(1);
 
-            const scheduleTemplate = `
+            let scheduleTemplate = [`
               {
                 "Type": "AWS::Events::Rule",
                 "Properties": {
                   "ScheduleExpression": "${ScheduleExpression}",
                   "State": "${State}",
                   "Targets": [{
                     "Arn": { "Fn::GetAtt": ["${normalizedFunctionName}LambdaFunction", "Arn"] },
-                    "Id": "${functionName}Schedule"
+                    "Id": "${functionName}Schedule"`,
+            ];
+            if (Input) {
+              scheduleTemplate.push(`,"Input": "${Input}"`);
+            }
+            if (InputPath) {
+              scheduleTemplate.push(`,"InputPath": "${InputPath}"`);
+            }
+            scheduleTemplate.push(`
                   }]
                 }
               }
-            `;
+            `);
+            scheduleTemplate = scheduleTemplate.join('');
```

Comment:
I am just thinking that the following might be a valid alternative, but even if more compact I am not totally sure it can be considered more readable:

```javascript
const template = `
  {
    "Type": "AWS::Events::Rule",
    "Properties": {
      "ScheduleExpression": "${ScheduleExpression}",
      "State": "${State}",
      "Targets": [{
        ${Input ? '"Input": "${Input}",' : ''}
        ${InputPath ? '"InputPath": "${InputPath}",' : ''}
        "Arn": { "Fn::GetAtt": ["${normalizedFunctionName}LambdaFunction", "Arn"] },
        "Id": "${functionName}Schedule"
      }]
    }
  }
`;
```